### PR TITLE
Bug 1182924 - Adds APN with multiple APN types correctly

### PR DESCRIPTION
**Pull request by scottwu**

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=b94114ee150bf9059a2fa4942fa1ea77bd801fd3

File: apps/settings/js/panels/apn_editor/apn_editor_session.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 /**
  * The apn editor session module
  */
-define(function() {
+define(function(require) {
```
Comment (gasolin) [inline apps/settings/js/panels/apn_editor/apn_editor_session.js]: nice catch :+1:

File: apps/settings/js/panels/apn_editor/apn_editor_session.js [hunk 2]
```diff
@@ -41,9 +41,10 @@ define(function() {
       var promises = [];
       var newApnSetting = this._exportApnSetting(this._inputElements);
       newApnSetting.types.slice().forEach(function(type) {
-        newApnSetting.types = [type];
+        var settingClone = ApnUtils.clone(newApnSetting);
```
Comment (gasolin) [inline apps/settings/js/panels/apn_editor/apn_editor_session.js]: :100:  
Please add some note here to help people know we should clone object here or the promise push will always take the last newApnSetting.types value

**Comment by gasolin** [comment]
please add `,r=gasolin` at the end of the commit

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=0a2059675e87b6b62cbbd70a7628a4fce22edf56

**Pull request merged**